drivers/interrupt_controller/intc_gicv3.c

```diff
@@ -54,15 +54,17 @@ void arm_gic_irq_set_priority(unsigned int intid,
 	sys_write8(prio & GIC_PRI_MASK, IPRIORITYR(base, intid));
 
 	/* Interrupt type config */
-	idx = intid / GIC_NUM_CFG_PER_REG;
-	shift = (intid & (GIC_NUM_CFG_PER_REG - 1)) * 2;
-
-	val = sys_read32(ICFGR(base, idx));
-	val &= ~(GICD_ICFGR_MASK << shift);
-	if (flags & IRQ_TYPE_EDGE) {
-		val |= (GICD_ICFGR_TYPE << shift);
+	if (!gic_is_sgi(intid)) {
+		idx = intid / GIC_NUM_CFG_PER_REG;
+		shift = (intid & (GIC_NUM_CFG_PER_REG - 1)) * 2;
+
+		val = sys_read32(ICFGR(base, idx));
+		val &= ~(GICD_ICFGR_MASK << shift);
+		if (flags & IRQ_TYPE_EDGE) {
+			val |= (GICD_ICFGR_TYPE << shift);
+		}
+		sys_write32(val, ICFGR(base, idx));
```

Comment:
SGI `ICFGR`s are specified to be RAO/WI, so this is technically not necessary (we can safely assume that any writes to SGI ICFGRs will be ignored and have no effect).

Reply:
> Please fix the build failure with `qemu_cortex_r5`.

I will check this. I am not sure what happens with Cortex-R w.r.t default priority.

Reply:
> SGI `ICFGR`s are specified to be RAO/WI, so this is technically not necessary (we can safely assume that any writes to SGI ICFGRs will be ignored and have no effect).

lets have these explicit checks, that will be a sort of documentation of the fact that such configs are not applicable for SGI.